user -> develop

ogh010 opened the pull request:
## #️⃣연관된 이슈

> #42 #49 

## 📝작업 내용

> 회원 정보 수정, 탈퇴 기능 
상점 페이징 처리
상점 검색어 기반 검색기능
상점 찜 추가,삭제

### 스크린샷 (선택)


## 💬리뷰 요구사항(선택)

>

dlsundn committed:
Refactor : getStoreList함수의 반환값 void로 변경

dlsundn committed:
Merge remote-tracking branch 'origin/develop' into sunwoo/user/store/list

sngsngUDON committed:
review Test

sngsngUDON committed:
Store.jsp js/css/jsp 파일로 분리 및 store 기능 부분 개선

1. Store.jsp 에 있었던 Javascript 부분과 css 부분을 파일 분리했습니다.
2. Store 부분의 부족했던 기능을 개선

sngsngUDON committed:
md파일 삭제(따로 기록)

dlsundn committed:
Merge remote-tracking branch 'origin/minseok/user/review' into sunwoo/user/store/like

# Conflicts:
#	src/main/java/kr/co/solfood/user/store/StoreController.java

ogh010 committed:
feat : join -> register 로 변경

ogh010 committed:
feat : 소문자로 변경

ogh010 committed:
feat : 마이페이지 수정 진행중

sngsngUDON committed:
들여쓰기 수정 및 BEAN 주입 에러 해결

sngsngUDON committed:
인터셉터 파일 수정 및 에러 발생시 페이지 이동 통일

1. User 로그인이 됐을 때만 리뷰 작성, 수정, 삭제를 할 수 있도록 인터셉터 파일 변경했습니다.
2. 카테고리 설정 정보 API 를 AdminHomeController 로 이동했습니다.
3. review 관련 url 매핑을 수정했습니다.

sngsngUDON committed:
파일명 변경, url 매핑 변경

1. reviewList.jsp 를 review 폴더 아래 list.jsp 로 변경했습니다. (중복이슈)
2. 카테고리 설정 정보 API 를 다시 storeController 로 옮겼습니다. (검색어 할 때 사용이 됩니다.)
3. 각 에러 상황에 맞게 url 매핑을 개선했습니다.

sngsngUDON committed:
Merge branch 'user' into minseok/user/review

sngsngUDON committed:
MvcConfig파일 수정

5y1ee committed:
Merge pull request #48 from SHDS-ESC/minseok/user/review

review Test 초안입니다.

ogh010 committed:
충돌
Merge branch 'user' of https://github.com/SHDS-ESC/Sol-Food into gahee/user/login/native-login

# Conflicts:
#	src/main/java/configuration/MvcConfig.java

ogh010 committed:
feat:import 수정

sngsngUDON committed:
Aws S3 연동 및 main 페이지 개선

AWS S3 연동 성공했고, 메인 페이지 개선했습니다.

sngsngUDON committed:
AWS S3 성능개선

1. FileUploadController.java 에서 보면 아무런 인증 없이 누구나 매핑된 url로 접근할 수 있는 문제점을 발견
-> 회원가입 진행 중인 세션에서만 접속 가능하도록 개선 + IP 별로 분당 요청 수 제한을 걸었습니다.

2. 적용된 보안 검증들
1. 세션 검증 → joinInProgress 플래그 확인
2. 업로드 횟수 제한 → 세션당 5회
3. Rate Limiting → IP별 분당 10회
4. 파일 확장자 검증 → 이미지만 허용
5. 파일 크기 제한 → JavaScript에서 5MB

❌ 문제 1: 불필요한 API 호출 구조
🔴 3단계 호출 - 비효율적!
1. /profile/upload-url → Pre-signed URL 받기
2. S3에 직접 업로드
3. /profile/complete → 공개 URL 받기
문제점: 3번째 호출이 불필요함! S3 URL은 예측 가능하므로 클라이언트에서 바로 생성 가능

❌ 문제 2: AWS SDK v1 사용 (성능 이슈)
현재: AWS SDK v1.12.500 사용
개선: AWS SDK v2로 업그레이드하면 성능 30% 향상 + 더 적은 메모리 사용

sngsngUDON committed:
Merge branch 'user' into minseok/user/awsS3

dlsundn committed:
feat : 페이징 처리를 위한 util class 생성

dlsundn committed:
refactor : 페이징 처리를 위해 pagedto 상속

dlsundn committed:
feat : 더보기 페이징 처리 구현

dlsundn committed:
feat : 더보기 페이징 처리 구현

ajax로 데이터를 받아 보여주는 방식

sngsngUDON committed:
Merge pull request #54 from SHDS-ESC/minseok/user/awsS3

AWS 연동

ogh010 committed:
Merge branch 'user' into gahee/user/login/native-login

# Conflicts:
#	src/main/java/kr/co/solfood/user/login/LoginController.java
#	src/main/java/kr/co/solfood/user/store/StoreController.java

ogh010 committed:
feat : join -> register로 변경

sngsngUDON committed:
Merge pull request #60 from SHDS-ESC/gahee/user/login/native-login

Gahee/user/login/native login

dlsundn committed:
Merge remote-tracking branch 'origin/user' into sunwoo/user/store/likes

# Conflicts:
#	src/main/java/kr/co/solfood/user/store/StoreController.java

sngsngUDON committed:
Merge pull request #61 from SHDS-ESC/sunwoo/user/store/likes

store 더보기 기능 구현

sngsngUDON committed:
검색어 기반 검색 기능 추가

검색 기능 (미완)

sngsngUDON committed:
Merge branch 'user' into minseok/user/awsS3

sngsngUDON committed:
fileUploadSessionInterceptor 추가 및 검색어 기반 검색 기능 추가

1. 원래 따로 하던 세션 검증을 fileUploadSessionInterceptor 파일을 새로 만들어 원래 있던 인터셉터에 추가했습니다.

2. 제가 작업한 검색어 기반 가게 검색 기능과 선우가 작업한 더보기 기능을 합쳐서 개선했습니다.

ogh010 committed:
feat : 파일업로드관련 빈 의존성 주입

ogh010 committed:
feat : 마이페이지 > 내정보 수정 완료

ogh010 committed:
feat: 탈퇴하기 구현

sngsngUDON committed:
Merge pull request #62 from SHDS-ESC/minseok/user/awsS3

AWS 인터셉터 추가와 store 기능 병합

sngsngUDON committed:
Category 설정 로직 개선 및 파일 구조 개선

저도 최대한 노력해서 정리하긴 했는데 아직 개선해야할 부분도 있을 것 같아서 피드백 주시면 감사하겠습니다.

sngsngUDON committed:
1차 리뷰 개선

1. 상수가 3곳에 중복 정의돼서 ReViewConstants 파일에 정리했습니다.
2. 지원님이 리뷰해주신 부분 개선

5y1ee committed:
Merge pull request #63 from SHDS-ESC/gahee/user/login/native-login

Gahee/user/login/native login

sngsngUDON committed:
Merge branch 'user' into minseok/user/awsS3

5y1ee committed:
fix : s3 업로드 함수 오류 수정 (최적화 적용 이전 버전 롤백)

@sngsngUDON
최적화 시도는 좋으나 수정 이후 테스트 반드시 해볼 것.

5y1ee committed:
fix : mypage 수정 시 인터셉터에서 막히던 버그 수정

@ogh010
기능 추가 시 왜 추가했는지 이유를 좀 더 생각해보세요.

sngsngUDON committed:
최종 커밋

마이페이지에서 로그아웃이랑 수정된 닉네임 정보가 업데이트되지 않는 부분을 수정했습니다.

sngsngUDON committed:
Merge pull request #66 from SHDS-ESC/minseok/user/awsS3

Category 설정 로직 개선 및 파일 구조 개선

5y1ee commented on src/main/java/config/MvcConfig.java:
```diff
@@ -83,13 +88,26 @@ public AdminLoginInterceptor adminLoginInterceptor() {
+                .excludePathPatterns("/user/userControl/search-id")
+                .excludePathPatterns("/user/userControl/kakaoLogin")
+                .excludePathPatterns("/user/userControl/extra")
+                .excludePathPatterns("/user/store/**");         // 상점 목록/상세는 로그인 없이 접근 가능
```
이 패키지 구조를 처음 봤는데 몇가지 마음에 들지 않는게 있습니다.
1. 왜 user 밑에 userControl 이란 경러를 뒀는가? 굳이 userControl 이란 경로가 필요한 이유가 뭐죠?
2. userControl 이 컨벤션 상 옳은가? 우리가 정해두진 않았지만, 케멀 케이스를 경로에 쓰는건 어색해보입니다. native-login 처럼 user-control 이 낫지 않을까 싶은. (그러고 보니 kakaoLogin도 그렇게 되어있네요. kakao-login 이 더 좋게 보입니다 저는.)
3. 사실상 userControl 밑의 거의 모든 api 들이 exclude 됐는데 **로 모두 제외해도 되지 않는지?
4. userControl은 사실상 LoginController 아래에 있는 api 던데, user/login/login 같은 구조를 막고 싶었던거죠? 차라리 그렇다면 기존 login 경로를 (/user/userControl/login) main, home, page 등 다른 경로로 바꾸는게 낫지 않을지? login 이란 단어를 LoginController의 경로로 양보하되 실제 login의 메인 페이지는 login/main이나 login/home 등이 좋지 않을까요?

sngsngUDON replied:
아시다시피 원래 유저 밑에 user 회원 관리jsp 들과 store, review 관련된 모든 jsp들이 섞여있었습니다. 그래서 기능 별로 파일을 나누다가 회원관리 쪽도 파일로 나누게 됐습니다. 제가 캐멀 케이스에 좀 절여져있어서 아무 생각없이 이름을 지은 것 같긴 합니다 ㅎㅎ;

5y1ee commented on src/main/java/config/PropertiesConfig.java:
```diff
@@ -36,10 +37,22 @@ public ServerProperties serverProperties(
     public KakaoProperties kakaoProperties(
             @Value("${kakao.restApiKey}") String restApiKey,
-            @Value("${kakao.jsApiKey") String jsApiKey
+            @Value("${kakao.js.key}") String jsApiKey
```
이건 변수명 대신 application.properties 의 값을 변경하는게 좋겠다고 리뷰한 바 있는 것 같은데, restApiKey처럼 jsApiKey 가 낫지 않을까요?

sngsngUDON replied:
네네 통일해서 쓰는게 더 좋아보입니다

5y1ee commented on src/main/java/kr/co/solfood/admin/crawler/StoreWebCrawler.java:
```diff
@@ -140,6 +146,10 @@ private List<StoreVO> parseKakaoResponse(String jsonResponse, String category) {
+                    
+                    // 카테고리 ID 설정
+                    Integer categoryId = categoryService.getCategoryIdByName(standardCategory);
+                    store.setCategoryId(categoryId);
```
이 파일 자체는 admin 쪽인데 user에서 작업한 이유는?
카테고리 id가 null인 경우가 있어서 Integer로 한건가요? 그렇다면 어떤 경우에 카테고리 id가 null 일까요?

sngsngUDON replied:
사실 제가 store 쪽 작업하면서 가게 정보들이 어느 정도 충분히 있어야 테스트하기도 쉬울 것 같다고 생각해서 크롤러 부분을 추가했습니다. 만들고 보니 크롤러 부분은 admin 만 사용해야 맞다고 생각해서 admin 으로 배정했습니다. 
null 이 나올 가능성이 낮긴 하지만 데이터베이스 연결이 안되어있거나 CategoryProperties 에서 전달되는 카테고리 명이 null 으로 넘어올 수 도 있습니다.

5y1ee commented on src/main/java/kr/co/solfood/admin/home/AdminHomeController.java:
```diff
@@ -1,13 +1,17 @@
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
```
여긴 소스에 아무 변화도 없는데 import 문만 추가됐네요?
대부분 쓰지 않는 클래스일텐데, 이런 부분은 커밋하기 전에 확인하고 지워도 되지 않을까요?

5y1ee commented on src/main/java/kr/co/solfood/common/s3/FileUploadController.java:
```diff
@@ -0,0 +1,177 @@
+    @Autowired
+    private RateLimitService rateLimitService;
+    
+    // 세션 검증 로직은 FileUploadSessionInterceptor로 이동
```
필요없는 주석 지워주세요.

5y1ee commented on src/main/java/kr/co/solfood/common/s3/S3ServiceV2.java:
```diff
@@ -0,0 +1,147 @@
+    /**
+     * 🚀 Pre-signed URL 생성 (성능 최적화)
+     */
+    // public String generatePresignedUploadUrl(String fileName) {
```
이 주석 지웁시다. 아마 제가 남긴 것 같은데, @sngsngUDON 이 지워주세요.
그리고 s3 같은 외부 라이브러리 사용하는 코드는 변화가 있으면 반드시 테스트 해주세요.
저도 코드 리뷰를 하지만, 외부 코드는 보기만 해서 알 수 없기 때문에 테스트가 반드시 필요합니다.
테스트 코드를 만드는게 베스트겠지만요.

sngsngUDON replied:
네 확인했습니다. 이런 부분들 고칠 때에는 테스트 반드시 해보겠습니다.

5y1ee commented on src/main/java/kr/co/solfood/user/category/CategoryVO.java:
```diff
@@ -0,0 +1,10 @@
+import lombok.Data;
+
+@Data
+public class CategoryVO {
```
카테고리가 user 패키지 내부에 있는게 불편합니다.
category, store, menu는 지난주에 회의했던대로 user, owner, admin 과 같은 레벨의 패키지로 둡시다.

5y1ee replied:
#29 에 이슈가 있는데 트래킹이 안되고 있었군요,
제가 작업하던가 해야겠네요.

5y1ee commented on src/main/java/kr/co/solfood/user/category/CategoryServiceImpl.java:
```diff
@@ -0,0 +1,76 @@
+            CategoryVO category = categoryMapper.selectCategoryByName(categoryName.trim());
+            if (category == null) {
+                log.warn("카테고리명 '{}'에 해당하는 카테고리가 존재하지 않습니다. 기본 카테고리 ID(1) 반환", categoryName);
+                return 1; // 기본값: 전체 카테고리
```
예외처리는 좋은데 이런 값도 Constant 클래스에 두면 좋습니다. 만약 공통 패키지용 Constant 클래스가 없어서 이거 하나 때문에 만들기 좀 그렇다면, 이 클래스에 상수로 두면 됩니다.

5y1ee commented on src/main/java/kr/co/solfood/user/login/LoginController.java:
```diff
@@ -14,7 +14,7 @@
 
 @Controller
-@RequestMapping("/user")
+@RequestMapping("/user/userControl")
```
이 매핑을 다시 생각해봅시다. (맨 처음 리뷰 참고)

sngsngUDON replied:
넵 이 부분에 대해서는 저 혼자 생각하고 성급하게 넣었던 것 같아서 같이 논의를 해보는 것이 좋을 것 같습니다.

5y1ee commented on src/main/java/kr/co/solfood/user/login/LoginController.java:
```diff
@@ -117,24 +117,35 @@ public String searchPwd(SearchPwdRequest req, Model model) {
+        // 회원가입 진행 세션 플래그 설정 (S3 업로드 보안용)
+        session.setAttribute("joinInProgress", true);
+        session.setAttribute("uploadCount", 0);
+        session.setMaxInactiveInterval(30 * 60); // 30분 후 만료
```
이런 값도 constant 값 저장하는 클래스에 넣어서 관리합시다.

5y1ee commented on src/main/java/kr/co/solfood/user/mypage/MypageServiceImpl.java:
```diff
@@ -0,0 +1,25 @@
+    @Override
+    public boolean withdrawUser(long usersId) {
+       int result = mypageMapper.updateStatusToWithdraw(usersId);
+       return result == 1; // 1건 수정되면 성공
```
1과 비교하는 이유가 있을까요? boolean으로 처리하면 되지 않을지?
혹은 복수의 데이터가 변환될 걸 우려해서 이렇게 한건가요? 그렇다면 예외처리 로직이 있어야 할 것 같은데 그런건 보이지 않는군요.

ogh010 replied:
예외처리 로직은 컨트롤러에서 리턴받아서 사용합니다. 
1 과 0 을 성공, 실패로 보고 리다이렉트를 다르게 합니다.

5y1ee commented on src/main/java/kr/co/solfood/user/review/ReviewConstants.java:
상수용 클래스 좋습니다. 패키지 별로 모아둔 것도 좋은 방법인 것 같습니다.
여기서 더 나아가서, 더 상위 레벨의 패키지에서도 이런 클래스를 둬서 session에서 쓰이는 변수 (ex. userLoginSession, joinInProgress 등) 들도 이런 클래스에 변수로 두는게 좋아 보입니다.
이 부분 @sngsngUDON 이 작업해보시는건 어떨까요?

sngsngUDON replied:
넵 주말에 전체 페이지 검토해보면서 한번 만들어보겠습니다.

5y1ee commented on src/main/java/kr/co/solfood/user/review/ReviewController.java:
```diff
@@ -13,177 +12,127 @@
+    public String reviewWrite(@ModelAttribute ReviewVO review, RedirectAttributes redirectAttributes) {
         try {
+            if (!reviewService.isValidStarRating(review.getReviewStar())) {
+                redirectAttributes.addFlashAttribute("error", "올바른 별점을 선택해주세요.");
```
constant 클래스도 있겠다. 이런 리터럴도 변수로 씁시다. (아래에 있는 모든 코드 전부 다)

5y1ee commented on src/main/java/kr/co/solfood/user/review/ReviewController.java:
```diff
@@ -13,177 +12,127 @@
     private KakaoProperties kakaoProperties;
     
-    // 한 가게의 리뷰 목록 페이지 (storeId로 필터)
-    @GetMapping("/list")
```
user/review/list 경로는 아예 삭제된건가요?

sngsngUDON replied:
원래 저 매핑이 가게 상세 페이지로 가는 것이었는데 너무 맞지 않는 것 같아서 store/detail 로 변경했습니다.

5y1ee commented on src/main/java/kr/co/solfood/user/review/ReviewController.java:
리뷰 관련 페이지를 실제로 확인해보고 싶은데, 매핑된 버튼이 없어 보입니다. 이제 슬슬 프로젝트도 중반을 향해 가는데 임시 버튼으로 연결해두면 테스트 하면서 리뷰하는데 큰 도움이 될 것 같습니다.
(제가 못 찾은 것 일 수도 있습니다.)

sngsngUDON replied:
리뷰 관련 페이지는 리뷰 작성하는 reviewWrite 와 가게 상세페이지에서 리뷰 메뉴 클릭하면 뜨는 페이지 이 2개 뿐이긴 합니다.
리뷰 작성 페이지는 user/reivew/write 로 현재 접속 가능하고, 결제 내역 페이지와 연결하면 될 것 같습니다! 그 전까지 쓰기위해 메인 페이지에 버튼 하나 만들어 놓겠습니다.

sngsngUDON replied:
<img width="371" alt="image" src="https://github.com/user-attachments/assets/b3c8d767-b993-4a3a-ade0-3a176c9eff79" />

5y1ee commented on src/main/java/kr/co/solfood/user/review/ReviewService.java:
```diff
@@ -6,6 +6,12 @@
 public interface ReviewService {
+    
+    // === 상수 정의 ===
+    int MIN_STAR_RATING = 1;
```
상수 클래스에 있는건데 지워주시죠.

5y1ee commented on src/main/java/kr/co/solfood/user/review/ReviewServiceImpl.java:
```diff
@@ -22,80 +23,108 @@ public class ReviewServiceImpl implements ReviewService {
     
     @Override
+    @Transactional
     public boolean registerReview(ReviewVO review) {
```
registReview 가 올바르지 않을까 생각했는데, 알고보니 regist 라는 단어는 없고 register 가 동사로 등록하다 라는 뜻이로군요; 오히려 명사형은 registry...

5y1ee commented on src/main/java/kr/co/solfood/user/review/ReviewServiceImpl.java:
```diff
@@ -22,80 +23,108 @@ public class ReviewServiceImpl implements ReviewService {
-                .filter(r -> r.getReviewTitle() != null && r.getReviewTitle().contains(reviewTitle))
-                .collect(Collectors.toList());
+        if (restaurantName == null || restaurantName.trim().isEmpty()) {
+            throw new IllegalArgumentException("음식점 이름은 비어있을 수 없습니다.");
```
리터럴 대신 변수로.
그리고 여긴 try catch를 쓰지 않은 이유라도?

sngsngUDON replied:
딱히 안 쓸 이유는 없는 것 같습니다. try-catch 문으로 통일 시키는 것이 좋아 보입니다.

5y1ee commented on src/main/java/kr/co/solfood/user/review/ReviewVO.java:
```diff
@@ -1,22 +1,71 @@
 public class ReviewVO {
+    
+    // === 상수 정의 ===
+    public static final String STATUS_ACTIVE = "Y";
```
이것도 지워주세요.

5y1ee commented on src/main/java/kr/co/solfood/user/review/ReviewValidator.java:
```diff
@@ -0,0 +1,98 @@
+     * @param review 검증할 리뷰 객체
+     * @throws IllegalArgumentException 유효성 검증 실패 시
+     */
+    public static void validateReview(ReviewVO review) {
```
구조가 보기 좋습니다.

5y1ee commented on src/main/java/kr/co/solfood/user/review/ReviewServiceImpl.java:
```diff
@@ -22,80 +23,108 @@ public class ReviewServiceImpl implements ReviewService {
+    @Transactional
     public boolean updateReview(ReviewVO review) {
+        validateReview(review);
+        validateReviewId(review.getReviewId());
```
하나하나 다시 살펴보니, 굉장히 부자연스러워 보입니다.
validateReview 다음으로 validateReviewId가 있다면 도대체 이 함수를 들어가지 않고서는 뭘 하는건지 감이 안 옵니다.

validate 코드를 남발했는데, 이해는 전혀 못하겠습니다. 그래서 잘 안 읽혀서 처음엔 대충 넘겨봤나봅니다.;;

5y1ee commented on src/main/java/kr/co/solfood/user/review/ReviewValidator.java:
```diff
@@ -0,0 +1,98 @@
+    /**
+     * 가게 ID 유효성 검증
+     */
+    public static void validateStoreId(Integer storeId) {
```
validateReviewNotNull은 null 비교만 있어도 오케이. 이름이 그러니까.
근데 validateStoreId랑 아래 다른 메서드는 왜 null 검사만 하죠?

validate 관련해서 대대적으로 손을 좀 봐야겠습니다.

sngsngUDON replied:
네 맞는 말입니다. 유효성 검증이 너무 단순한 것 같아 이건 좀 고쳐야 할 것 같네요

5y1ee commented on src/main/java/kr/co/solfood/user/store/CategoryProperties.java:
Properties 의 의미가 뭘까요?
루트 패키지에도 properties가 존재하는데 카테고리 프로퍼티 클래스는 어느 패키지에 있는게 올바르다고 생각하시나요?

5y1ee replied:
심지어 category 패키지도 아니고 store 패키지에 있었네요..?

sngsngUDON replied:
원래 카테고리 테이블 생성하기 전에 이곳에서 카테고리 관련 작업들을 하는 파일로 만들었는데 테이블 만든 이후로는 솔직히 필요성을 잘 못느끼겠네요,, category 패키지가 table 새로 생성하면서 새로 만들어진거라 이 파일을 옮긴다는 것을 잊었나봅니다..

5y1ee commented on src/main/java/kr/co/solfood/user/store/CategoryProperties.java:
```diff
@@ -17,34 +24,98 @@ public class CategoryProperties {
     static {
         // 검색 키워드 초기화
-        SEARCH_KEYWORDS.put("전체", "음식점");
+        SEARCH_KEYWORDS.put(StoreConstants.CATEGORY_ALL, StoreConstants.DEFAULT_SEARCH_KEYWORD);
```
이 수 많은 put을 보고 무슨 생각을 했나요?
혹시 static 블럭을 쓴 이유가 있을까요? (알고 쓴건지 질문입니다. 검색해서 그러그러하다. 고 대답하라고 하는 말 아닙니다.)

sngsngUDON replied:
static 블럭에 대해는 솔직히 잘 모르겠네요, 솔직히 비효율적일 것 같다고 생각하긴 했습니다. 일단 다른 기능들을 먼저 한다 생각하고 뒷전에 미뤄 놨는데 다시 보니까 테이블을 만들어서 관리 하는 것이 훨씬 좋은 것 같습니다.

sngsngUDON replied:
이 파일은 삭제하고 categoryService 파일로 필요한 부분만 병합 예정입니다.

5y1ee commented on src/main/java/kr/co/solfood/user/store/CategoryProperties.java:
```diff
@@ -17,34 +24,98 @@ public class CategoryProperties {
+    }
+    
+    // 데이터베이스에서 카테고리 목록을 실시간으로 가져오기
+    private List<CategoryVO> dbCategories = new ArrayList<>();
```
얘는 그냥 필드네요?

이 경우 arraylist 객체는 2개가 존재하는 순간이 생기고, 위의 상수로 선언한 map과는 달리 이 list는 변하지 않음을 보장하진 않겠군요.

5y1ee commented on src/main/java/kr/co/solfood/user/store/CategoryProperties.java:
```diff
@@ -17,34 +24,98 @@ public class CategoryProperties {
-        return new ArrayList<>(SEARCH_KEYWORDS.keySet());
+        try {
+            // CategoryService가 사용 가능할 때만 데이터베이스에서 조회
+            if (categoryService != null) {
```
autowired 필드인데 null 체크가 필요할까요?

5y1ee commented on src/main/java/kr/co/solfood/user/store/StoreController.java:
이 컨트롤러를 보고 경악을 금치 못했습니다.

1. 분명 가게 컨트롤러인데 연관있는 서비스 객체는 죄다 필드로 넣어놨군요.
2. ResponseVO 는 왜 여기에 다 선언해놨나요?

이런 코드는 작성자 외에는 전혀 디버깅을 할 수 없는 코드 구조로 보입니다. StoreController 가 너무 많은 역할을 갖고 있어요. 프로젝트 진행을 위해. 리팩터링을 강요하진 않겠지만 앞으로 뭔가 문제가 나타나서 수정이 필요할 때 반드시 작성한 사람이 수습을 해야 할 것 같습니다.

sngsngUDON replied:
2번은 저희도 확인하고 분리하려고 했는데 이번 병합에 안 들어갔네요,, 이번에 분리하겠습니다. StoreController를 두명이 같이 작업하다 보니 이것저것 같이 섞인 것 같습니다. 이 부분 좀 개선해보겠습니다.

5y1ee commented on src/main/java/kr/co/solfood/user/store/StoreController.java:
```diff
@@ -1,69 +1,501 @@
+        
+        // 별점별 개수를 배열로 변환 (1점부터 5점까지)
+        long[] starCounts = new long[STAR_COUNT];
+        if (starCountsMap != null) {
```
분명 지난 pr에서 for문으로 개선하라는 리뷰를 본 것 같은데 반영 안했나요?

sngsngUDON replied:
확인하고 반영하겠습니다

5y1ee commented on src/main/resources/kr/co/solfood/user/category/CategoryMapper.xml:
```diff
@@ -0,0 +1,37 @@
+        SELECT 
+            category_id,
+            category_name,
+            category_image
```
그냥 * 때리면 되지 않는지?

sngsngUDON replied:
그래도 되기는 합니다 그래도 어떤 데이터를 조회하는지 쉽게 알아볼 수 있으면 좋지 않을까요 ??

5y1ee commented on src/main/resources/kr/co/solfood/user/mypage/MypageMapper.xml:
```diff
@@ -0,0 +1,29 @@
+            users_email = #{usersEmail},
+            users_nickname = #{usersNickname},
+            users_name = #{usersName},
+            users_birth = #{usersBirth},
```
수정 페이지 들어가면 생년월일 초기화 되어 있던데 그거 수정해주시죠. (ux상 불편함)

ogh010 replied:
네

5y1ee commented on src/main/resources/kr/co/solfood/user/review/ReviewMapper.xml:
```diff
@@ -66,80 +70,124 @@
+            #{reviewCommentId}, 
+            #{reviewStar}, 
+            #{reviewImage}, 
+            CURRENT_TIMESTAMP, 
```
이건 DB에서 지금 시간 알아서 넣으라는거죠?
java에서 객체를 만들지 않아도 되니까, 이것도 좋은 방법 같네요.

sngsngUDON replied:
네네 db에서 따로 처리하는게 좋아보입니다.

5y1ee commented on src/main/webapp/WEB-INF/views/owner/home.jsp:
```diff
@@ -352,7 +352,7 @@
                 <li><a href="#about">소개</a></li>
                 <li><a href="#contact">문의</a></li>
-                <li><a href="/user/login">로그인</a></li>
+                                        <li><a href="/user/userControl/login">로그인</a></li>
```
어쩌다가 탭을 이렇게..

5y1ee reviewed (changes requested):
XML, JSP, JS, test 코드는 제가 일일이 체크 못했습니다.

회의 때 더 얘기해봅시다.

jiwonns reviewed (commented):
확인했습니다

sngsngUDON committed:
review 개선점(1)

1. js.key -> jsApiKey 로 변경
2. ReviewConstants,java 상수 클래스 제작 후 review 관련 파일에서 쓰던 상수들 다 정리했습니다.
3. CategoryProperties 에서 카테고리 테이블 생기면서 안쓰는 것들 다 없애고 categoryService 와 ServiceImpl 로 병합시켰습니다.
4. storeController 파일이 너무 기능이 많은 것 같아 파일을 좀 나눠봤습니다. 더 개선점 있으면 말해주십쇼

5y1ee committed:
CustomException 추가

5y1ee committed:
Merge pull request #75 from SHDS-ESC/minseok/user/changeUser

review 개선점(1)

sngsngUDON committed:
userControl -> login 폴더명 변경

sngsngUDON committed:
kakaoLogin -> kakao-login

sngsngUDON committed:
login.jsp -> loginPage.jsp

sngsngUDON committed:
kakao-login redirect 경로 수정

5y1ee committed:
예외 처리 예시 추가

5y1ee committed:
Merge pull request #76 from SHDS-ESC/minseok/user/changeUser

userControl -> login 폴더명 변경

5y1ee committed:
Merge branch 'user' into sangyoon/user/CustomException

jiwonns committed:
CustomException 추가 부분 병합

CustomException 추가 병합
